pulsar-broker/src/main/java/org/apache/pulsar/broker/storage/ManagedLedgerStorage.java

```diff
@@ -52,25 +57,25 @@ void initialize(ServiceConfiguration conf,
                     OpenTelemetry openTelemetry) throws Exception;
 
     /**
-     * Return the factory to create {@link ManagedLedgerFactory}.
-     *
-     * @return the factory to create {@link ManagedLedgerFactory}.
+     * Get all configured storage class instances.
+     * @return all configured storage class instances
      */
-    ManagedLedgerFactory getManagedLedgerFactory();
+    Collection<ManagedLedgerStorageClass> getStorageClasses();
 
     /**
-     * Return the stats provider to expose the stats of the storage implementation.
-     *
-     * @return the stats provider.
+     * Get the default storage class.
+     * @return default storage class
      */
-    StatsProvider getStatsProvider();
+    default ManagedLedgerStorageClass getDefaultStorageClass() {
+        return getStorageClasses().stream().findFirst().get();
+    }
 
     /**
-     * Return the default bookkeeper client.
-     *
-     * @return the default bookkeeper client.
+     * Lookup a storage class by name.
+     * @param name storage class name
+     * @return storage class instance, or empty if not found
      */
-    BookKeeper getBookKeeperClient();
+    Optional<ManagedLedgerStorageClass> getManagedLedgerStorageClass(String name);
```

Comment:
This method is hard to understand. From the perspective of customizing a managed ledger storage, what I want to know is that when could this method be called and how could the `name` be.

It seems that the name comes from `ManagedLedgerConfig#storageClassName`. But how could this config be changed?

Reply:
It seems that it can only be set via the topic policies.

Reply:
This PR doesn't consider how the `ManagedLedgerStorage` implementation configures the default value. It's possible to override the storage class name at namespace or topic level in persistence policies.